src/models/feature_selection/shap_selector.py

```
@@ -0,0 +1,131 @@
+    5. For each original feature aggregate score = sum_over_latents(|corr| * latent_shap_importance)
+    6. Return top-k features by aggregated score
+    """
+    if x_enc_train.ndim != 2:
+        raise ValueError("x_enc_train must be 2D numpy array")
+
+    # 1) Train LightGBM surrogate
+    try:
+        model = LGBMRegressor(n_estimators=500, learning_rate=0.05, n_jobs=-1)
+        model.fit(x_enc_train, y_train)
+    except Exception as e:
+        logger.error(f"LightGBM training failed: {e}")
```

Ruff (0.14.1)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)